interfaces/builtin/serial_port.go

```diff
@@ -159,17 +159,34 @@ func (iface *serialPortInterface) AppArmorConnectedPlug(spec *apparmor.Specifica
 }
 
 func (iface *serialPortInterface) UDevConnectedPlug(spec *udev.Specification, plug *interfaces.Plug, plugAttrs map[string]interface{}, slot *interfaces.Slot, slotAttrs map[string]interface{}) error {
+	// For connected plugs, we use vendor and product ids if available,
+	// otherwise add the kernel device
+	hasOnlyPath := true
+	if iface.hasUsbAttrs(slot) {
+		hasOnlyPath = false
+	}
+
 	usbVendor, vOk := slot.Attrs["usb-vendor"].(int64)
-	if !vOk {
+	if !vOk && !hasOnlyPath {
 		return nil
 	}
 	usbProduct, pOk := slot.Attrs["usb-product"].(int64)
-	if !pOk {
+	if !pOk && !hasOnlyPath {
 		return nil
 	}
+
+	path, pathOk := slot.Attrs["path"].(string)
+	if !pathOk && hasOnlyPath {
+		return nil
+	}
+
 	for appName := range plug.Apps {
 		tag := udevSnapSecurityName(plug.Snap.Name(), appName)
-		spec.AddSnippet(udevUsbDeviceSnippet("tty", usbVendor, usbProduct, "TAG", tag))
+		if hasOnlyPath {
+			spec.AddSnippet(fmt.Sprintf("SUBSYSTEM==\"tty\", KERNEL==\"%s\", TAG+=\"%s\"", strings.TrimPrefix(path, "/dev/"), tag))
```

Comment:
I decided to store this off in a variable because the function is iterating through loops to determine if have USB attributes, and I didn't want to call that multiple times for performance reasons.